R/rlang.R

```diff
@@ -64,8 +64,15 @@ base_pkg_env <- NULL
     is_same_body <<- is_reference
   }
 
+  # glue depends on rlang so we can't detect it is installed right
+  # away. Instead we do it when glue has finished loading.
+  setHook(
```

Comment:
Does this work if glue has is already loaded?

Reply:
oh I confused myself into thinking that glue depended on rlang, but it doesn't.

I have extracted `defer_if_unloaded()` from the downstream dep check routine. It should work without side effects in all cases. Better use this function no matter the dependency structure, to avoid mistakes.